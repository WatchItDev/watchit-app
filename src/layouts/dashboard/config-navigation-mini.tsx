export const data = [
  {
    id: '1',
    title: 'explore',
  },
  {
    id: '12',
    title: 'Comunity',
  },
  {
    id: '123',
    title: 'Governance',
  },
  {
    id: '1234',
    title: 'Marketplace',
  },
  {
    id: '12345',
    title: 'Events',
  },
  {
    id: '123456',
    title: 'Achievements',
  },
]
