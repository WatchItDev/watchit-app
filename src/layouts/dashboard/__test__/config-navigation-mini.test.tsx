import { data } from '../config-navigation-mini';

describe('[UI: mini sidebar items array', () => {
  it('to match the snapshot', () => {
    expect(data).toMatchSnapshot();
  });

  it('contains 5 items', () => {
    expect(data).toHaveLength(5);
  });

  it('each item has required properties', () => {
    data.forEach((item) => {
      expect(item).toHaveProperty('id');
      expect(item).toHaveProperty('title');
      expect(item).toHaveProperty('icon');
      expect(item).toHaveProperty('badge');
      expect(item).toHaveProperty('bgColor');
    });
  });

  it('each item has a unique id', () => {
    const ids = data.map((item) => item.id);
    const uniqueIds = new Set(ids);
    expect(uniqueIds.size).toBe(ids.length);
  });

  it('each item has the correct badge', () => {
    data.forEach((item) => {
      expect(item.badge).toBe('Ideas for Builders');
    });
  });

  it('each item has the correct bgColor', () => {
    data.forEach((item) => {
      expect(item.bgColor).toBe('#4A34B8');
    });
  });
});
