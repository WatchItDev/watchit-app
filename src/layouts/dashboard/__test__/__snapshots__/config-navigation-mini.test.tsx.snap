// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[UI: mini sidebar items array > to match the snapshot 1`] = `
[
  {
    "badge": "Ideas for Builders",
    "bgColor": "#4A34B8",
    "icon": <IconHeadphones />,
    "id": "1",
    "title": "ListenIt",
  },
  {
    "badge": "Ideas for Builders",
    "bgColor": "#4A34B8",
    "icon": <IconPolaroidFilled />,
    "id": "12",
    "title": "CaptureIt",
  },
  {
    "badge": "Ideas for Builders",
    "bgColor": "#4A34B8",
    "icon": <IconBook />,
    "id": "123",
    "title": "ReadIt",
  },
  {
    "badge": "Ideas for Builders",
    "bgColor": "#4A34B8",
    "icon": <IconFolderBolt />,
    "id": "1234",
    "title": "StoreIt",
  },
  {
    "badge": "Ideas for Builders",
    "bgColor": "#4A34B8",
    "icon": <IconDeviceGamepad2 />,
    "id": "12345",
    "title": "PlayIt",
  },
]
`;
